src/header.ts

```diff
@@ -0,0 +1,359 @@
+import Common from 'ethereumjs-common'
+import * as utils from 'ethereumjs-util'
+import { BN } from 'ethereumjs-util'
+import { Blockchain, BlockHeaderData, BufferLike, NetworkOptions, PrefixedHexString } from './types'
+import { Buffer } from 'buffer'
+import { Block } from './block'
```

Comment:
This also seems weird to me, as it forces a cyclic dependency between `block.ts` and `header.ts`.  

We could change the methods to operate on `Header`s, but the blockchain module also operates on `Block`s, so that cyclic dependency would still be needed.